### DISCUSSION TITLE
Records inserted via Data Loader did not trigger rollup operation

### DISCUSSION THREAD

**Discussion by jdids247** [jamessimone/apex-rollup#542] [Q&A]
New Apex Rollup user, so bear with me if there is a known setting that fixes this.

We use this on the OpportunityLineItemSchedule object to get the MAX Schedule Date and it's revenue number. We then update the OpportunityLineItem record the schedule is tied to.

Everything is working great when the data in the OpportunityLineItemSchedule is updated or added from the GUI, but we just had a user bulk load in schedule data using Data Loader and it didn't fire off the rollup. **I can see the jobs completed in the Apex Jobs log**

We're using CMDT-based trigger on the OpportunityLineItemSchedule object to kick this off

Here's the CMDT
![image](https://github.com/jamessimone/apex-rollup/assets/110419431/f9b76934-b08b-4259-8ada-c4389d368aa8)

Here's the trigger
```
trigger ApexRollup_LastPeriodARR on OpportunityLineItemSchedule(after insert, after update, before delete, after delete, after undelete) {
  // after delete is required ONLY if your org does Account / Contact / Lead / Case merges
  Rollup.runFromTrigger();
  // etc. You can invoke the above from your handler if you have one
}
```

I hope I'm missing something obvious here.

**Comment by jamessimone** [comment]
@jdids247 bulk data loads should definitely invoke Apex triggers. It's hard to speculate on what might be happening without more info though. For example: has the org Rollup Setting been properly configured? Is there a user or profile level exclusion that would affect the user doing the data load?

Typically, one way to validate things would be to start a trace for that user under the Debug Log section, do a small data load (preferably with Is Rollup Logging Enabled set to true on the Rollup Control CMDT org default record) and ensure in the logs that are created that Apex Rollup is being invoked correctly, and that there aren't issues updating the parent-level records for that running user or something.

**Reply by jdids247** [reply]
That specific user and all other users are able to get the rollups to update directly when editing the triggering record directly in Salesforce.

Thanks for the suggestion, I'll take a look at logging.

Appreciate the support on this. We're excited to move forward with your solution.

**Reply by jamessimone** [reply]
Of course. I'll work with you every step of the way to understand what's going on, and help with anything that comes up that isn't working properly.
